torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1999,7 +2015,56 @@ def print_results(grouped, indent, result_fn):
 
 
 class ShapeEnv:
+    # This is a wrapper over the actual __init__ function.
+    #
+    # Where to add a new constructor parameter to ShapeEnv?
+    # =====================================================
+    # This __init__ function should be used only for parameters related to event recording.
+    # These are parameters that we don't wish to pass down the road to new ShapeEnv instances
+    # created from replaying events.
+    #
+    # If you wish to add a parameter to the constructor of ShapeEnv, unrelated to event
+    # recording, do so in the _init function.
     def __init__(
+        self, *,
+        should_record_events: Optional[bool] = None,
+        tracked_fakes: Optional[List[Any]] = None,
+        **kwargs
+    ) -> None:
+        self._init(**kwargs)
+
+        # Disable event recording when replaying.
+        kwargs["should_record_events"] = False
```

Comment:
How do you know you're replaying though, here?

Reply:
This new value will only be used in case we are replaying the construction of a `ShapeEnv`. Note that below this line, we input `kwargs` to the creation of a `ShapeEnvEvent`.